VReplication CLI: Change references in local example from v1 to v2 commands

rohit-nayak-ps opened the pull request:
V2 cli commands are now the default in the local examples. This PR updates the docs related to local example to reflect the new commands.

We should merge this *after* the 10.0 docs are archived and we release 11.0.

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

rohit-nayak-ps committed:
Change references in local example from v1 to v2 commands

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

hallaroo committed:
formatting change

Signed-off-by: hallaroo <marla@planetscale.com>

hallaroo reviewed (approved):
Added one indent to make the formatting stick when I make the older version PDFs. Other than that LGTM

deepthi reviewed (commented):
This looks fine, but should we replace the `V2 Commands` item in this list with a `V1 Commands` page?
https://vitess.io/docs/reference/vreplication/

rohit-nayak-ps committed:
Move v2 commands to main vreplication directory from the v2 directory and move v1 commands to a new v1 directory. Related markdown changes. Simulate indentation of sub-commands.

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

rohit-nayak-ps committed:
Update description of landing page

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

rohit-nayak-ps committed:
Add back v2 directory since a link to it is used in the deprecation message in code at the moment

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

rohit-nayak-ps commented:
> This looks fine, but should we replace the `V2 Commands` item in this list with a `V1 Commands` page?
> https://vitess.io/docs/reference/vreplication/

Moved the commands from. v2 commands to the main directory and v1 to its own directory. We also keep the v2 commands in its own directory since the v2 links are used in code at the moment in a deprecation message in 11.0. We can remove this in 12.0.

deepthi commented on content/en/docs/reference/vreplication/cancel.md:
```diff
@@ -0,0 +1,29 @@
+
+### Description
+
+`Cancel` can be used if a workflow was created in error or was misconfigured and you prefer to create a new workflow instead of fixing this one. Cancel can only be called if no traffic has been switched. It removes vreplication-related artifacts like rows from vreplication and copy_state in the side-car \_vt database, routing rules, and blacklisted tables from the topo and, by default, the target ...
```
```suggestion
`Cancel` can be used if a workflow was created in error or was misconfigured and you prefer to create a new workflow instead of fixing this one. Cancel can only be called if no traffic has been switched. It removes vreplication-related artifacts like rows from vreplication and copy_state tables in the side-car `_vt` database and routing rules and blacklisted tables from the topo and, by default, the target tables/shards from the target keyspace.
```

deepthi commented on content/en/docs/reference/vreplication/complete.md:
```diff
@@ -0,0 +1,52 @@
+### Description
+**Alert: This is a destructive command**
+
+`Complete` is used after all traffic has been switched. It removes vreplication-related artifacts like rows from vreplication and copy_state in the side-car \_vt database, routing rules, and blacklisted tables (for MoveTables) from the topo. Also, by default, the source tables (or source shards) are dropped.
```
```suggestion
`Complete` is used after all traffic has been switched. It removes vreplication-related artifacts like rows from vreplication and copy_state tables in the side-car `_vt` database and routing rules and blacklisted tables (for MoveTables) from the topo. By default, the source tables (or source shards) are also dropped.
```

deepthi commented on content/en/docs/reference/vreplication/create.md:
```diff
@@ -0,0 +1,180 @@
+</div>
+
+#### table_specs
+**optional**  one of `table_specs` or `-all` needs to be specified
```
I think calling this optional is confusing. We just merged a PR changing it in the previous version of docs.
https://github.com/vitessio/website/pull/785/files#diff-73926e6e43b0ee43f3c3205be25703cf3b835d17503ba8b1cd743c999875032fR81

deepthi commented on content/en/docs/reference/vreplication/create.md:
```diff
@@ -0,0 +1,180 @@
+###### Uses
+
+* Improve performance by picking a tablet in cells in network proximity with the target
+* To reduce bandwidth costs by skipping cells that are in different availability zones
```
```suggestion
* Reduce bandwidth costs by skipping cells that are in different availability zones
```

deepthi commented on content/en/docs/reference/vreplication/create.md:
```diff
@@ -0,0 +1,180 @@
+
+###### Uses
+
+* To reduce the load on master tablets by using REPLICAs or RDONLYs
```
Fine for now. We will need to cleanup all website doc refs to MASTER after we merge https://github.com/vitessio/vitess/pull/8511

deepthi commented on content/en/docs/reference/vreplication/create.md:
```diff
@@ -0,0 +1,180 @@
+</div>
+
+###### Uses
+* allows updating the rows in `_vt.vreplication` after MoveTables has setup the
```
```suggestion
* Allows updating the rows in `_vt.vreplication` after MoveTables has setup the
```

deepthi commented on content/en/docs/reference/vreplication/create.md:
```diff
@@ -0,0 +1,180 @@
+provides an easier way to create simpler Materialize workflows by first using
+MoveTables with auto_start false, updating the BinlogSource as required by your
+Materialize and then start the workflow.
+* changing the `copy_state` and/or `pos` values to restart a broken MoveTables workflow
```
```suggestion
* Changing the `copy_state` and/or `pos` values to restart a broken MoveTables workflow
```

deepthi commented on content/en/docs/reference/vreplication/create.md:
```diff
@@ -0,0 +1,180 @@
+projection clause to modify the values on the target. This
+provides an easier way to create simpler Materialize workflows by first using
+MoveTables with auto_start false, updating the BinlogSource as required by your
+Materialize and then start the workflow.
```
```suggestion
Materialize and then starting the workflow.
```

deepthi commented on content/en/docs/reference/vreplication/create.md:
```diff
@@ -0,0 +1,180 @@
+
+<div class="cmd">
+If true the source schema is copied to the target shards. If false, you need to create the tables
+before calling reshard.
```
```suggestion
before calling `Reshard`.
```

deepthi commented on content/en/docs/reference/vreplication/movetables.md:
```diff
@@ -3,179 +3,69 @@ title: MoveTables
 
-A comma-separated list of cell names or cell aliases. This list is used by VReplication to determine which
-cells should be used to pick a tablet for selecting data from the source keyspace.<br><br>
+MoveTables is an "umbrella" command. The `action` sub-command defines the operation on the workflow.
```
suggestion: top-level instead of umbrella??

deepthi commented on content/en/docs/reference/vreplication/v2/_index.md:
```diff
@@ -1,5 +1,5 @@
-description: New V2 MoveTables and Reshard vtctld commands (Experimental)
-weight: 1000
+title: VReplication V2 commands
+description: "VReplication V2 vtctl commands (experimental)"
```
vtctl or vtctld? Let's choose one to be consistent.

deepthi commented on content/en/docs/reference/vreplication/reversetraffic.md:
```diff
@@ -0,0 +1,64 @@
+
+<div class="cmd">
+
+For master tablets, ReverseTraffic first stops writes on the target master and waits for the replication to the source to
```
replication "from" the source?
I think source and target get confusing when we are talking about Reverse.

deepthi commented on content/en/docs/reference/vreplication/switchtraffic.md:
```diff
@@ -0,0 +1,78 @@
+### Description
+
+`SwitchTraffic` switches traffic forward for the tablet_types specified. This replaces the previous
+SwitchReads and SwitchWrites commands with a single one. It is also possible now to switch all traffic with just one command. Also, you can now switch replica, rdonly and primary traffic in any order: earlier you needed to first
```
```suggestion
SwitchReads and SwitchWrites commands with a single one. It is now possible to switch all traffic with just one command. Also, you can now switch replica, rdonly and primary traffic in any order: earlier you needed to first
```

deepthi commented on content/en/docs/user-guides/migration/move-tables.md:
```diff
@@ -442,14 +445,15 @@ After this step is complete, you should see an error (in Vitess 9.0 and later) s
+
   * Before Vitess 9.0, the the routing rules from the source keyspace to the target keyspace was not cleaned up.  The assumption was that you might still have applications that refer to the tables by their explicit `schema.table` designation, and you want these applications to (still) transparently be forwarded to the new location of the data.  When you are absolutely sure that no applications ar ...
-  * From Vitess 9.0 onwards, the routing rules from the source keyspace to the target keyspace are also cleaned up as part of the `DropSources` operation. If this is not the behavior you want, can choose to either delay the `DropSources` until you are sure the routing rules (and source data) are no longer required; or you can perform the same steps as `DropSources` manually.
+  * From Vitess 9.0 onwards, the routing rules from the source keyspace to the target keyspace are also cleaned up as part of the `Complete` operation. If this is not the behavior you want, can choose to either delay the `Complete` until you are sure the routing rules (and source data) are no longer required; or you can perform the same steps as `Complete` manually.
```
```suggestion
  * From Vitess 9.0 onwards, the routing rules from the source keyspace to the target keyspace are also cleaned up as part of the `Complete` operation. If this is not the behavior you want, you can choose to either delay the `Complete` until you are sure the routing rules (and source data) are no longer required; or you can perform the same steps as `Complete` manually.
```

deepthi reviewed (approved):
Really nice docs.
I made some suggestions on wording, otherwise LGTM.
Please feel free to merge once the review has been addressed.

rohit-nayak-ps committed:
Address review comments

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

rohit-nayak-ps commented:
> I made some suggestions on wording, otherwise LGTM.
> Please feel free to merge once the review has been addressed.

Addressed all review comments.
